egcg_core/clarity.py

```diff
@@ -181,17 +204,22 @@ def get_user_sample_name(sample_name, lenient=False):
     :param bool lenient: If True, return the sample name if no user sample name found
     :return: the user's sample name, None or the input sample name
     """
-    user_sample_name = get_sample(sample_name).udf.get('User Sample Name')
+    user_sample_name = lims_samples_info(sample_name).get('User Sample Name')
+    if not user_sample_name:
+        user_sample_name = get_sample(sample_name).udf.get('User Sample Name')
     if user_sample_name:
         return sanitize_user_id(user_sample_name)
     elif lenient:
         return sample_name
 
 
 def get_sample_sex(sample_name):
-    sample = get_sample(sample_name)
-    if sample:
-        return sample.udf.get('Sex') or sample.udf.get('Gender')
+    sex = lims_samples_info(sample_name).get('Sex') or lims_samples_info(sample_name).get('Gender')
+    if not sex:
+        sample = get_sample(sample_name)
+        if sample:
+            sex = sample.udf.get('Sex') or sample.udf.get('Gender')
```

Comment:
So what happens if sample does not exist in this case, and therefore the if statement is not executed?

Reply:
`sex` should be set to `None`